Tabs bottom border radius and separator UI

petemill opened the pull request:
Fix https://github.com/brave/brave-browser/issues/962

Brave design spec shows no bottom border radius on tabs.
This patching method, although not great, is simple compared to the amount of code which would need to be copied over and only slightly modified, since bottom border radius is calculated in the middle of a long drawing function.

Feedback appreciated.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Needed or QA/No-QA-Needed) to include the closed issue in milestone 

## Test Plan:


## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

petemill commented:
Rebasing and testing on latest master / C70

petemill commented:
This is ready for review now. @bbondy @bridiver I also tagged you for review in case you had an opinion on this new patching method (making the chromium namespace non-anonymous so that it can be accessed by the brave override class).

petemill commented:
@simonhong that is valid, but it puts some of our customized logic mixed in with the chromium source, whilst we may have some other logic in the brave modules. The method currently in this PR is a proposal to keep the logic changes in the brave modules to avoid confusion. If we still prefer putting the function in the patched file over de-anonymizing the namespace, then sure - I'll make that change.

simonhong commented on browser/ui/views/tabs/brave_tab.cc:
```diff
@@ -0,0 +1,58 @@
+                       gfx::Tween::IntValueBetween(opacity, SK_AlphaTRANSPARENT,
+                                                   SK_AlphaOPAQUE));
+  };
+  const int separator_radius = kSeparatorThickness;
```
`kSeparatorThickness/2` ?

simonhong commented on browser/ui/views/tabs/brave_tab.cc:
```diff
@@ -0,0 +1,58 @@
+#include "ui/gfx/scoped_canvas.h"
+
+using namespace chrome_browser_ui_views_tabs_tab_cc;
+
```
nit: empty line

simonhong commented:
@petemill , Yep, my suggestion is just another option :)

petemill commented:
@simonhong addressed all feedback now - thanks. Decided to keep in the de-anonymizing namespace option after speaking to other team members due to the less worse impact on future rebasing.

simonhong commented on browser/ui/views/tabs/brave_tab.cc:
```diff
@@ -0,0 +1,57 @@
+  flags.setColor(separator_color(separator_opacities.right));
+  canvas->DrawRoundRect(
+                     trailing_separator_bounds, scaled_separator_radius, flags);
+}
```
nit: last empty line.

simonhong commented on browser/ui/views/tabs/brave_tab.h:
```diff
@@ -0,0 +1,28 @@
+#include "ui/gfx/geometry/rect.h"
+
+namespace chrome_browser_ui_views_tabs_tab_cc {
+  const gfx::RectF ScaleAndAlignBounds(const gfx::Rect& bounds,
```
nit: indent

simonhong commented on browser/ui/views/tabs/brave_tab.h:
```diff
@@ -0,0 +1,28 @@
+}
+
+class BraveTab : public Tab {
+  public:
```
nit: indent

petemill commented:
0.55.x f1d811d6

bbondy commented:
Added merged/0.55.x label
